wandb/sdk/launch/_project_spec.py

```diff
@@ -172,6 +172,45 @@ def image_name(self) -> str:
             assert self.job is not None
             return wandb.util.make_docker_image_name_safe(self.job.split(":")[0])
 
+    def fill_macros(self, image: str) -> None:
+        """Substitute values for macros in resource arguments.
+
+        Certain macros can be used in resource args. These macros allow the
+        user to set resource args dynamically in the context of the
+        run being launched. The macros are given in the ${macro} format. The
+        following macros are currently supported:
+
+        ${wandb_project} - the name of the project the run is being launched to.
+        ${wandb_entity} - the owner of the project the run being launched to.
+        ${wandb_run_id} - the id of the run being launched
+        ${wandb_run_name} - the name of the run being launched
+        ${wandb_image} - the URI of the docker image being launched for this run
```

Comment:
Well drat, I didn't use snake case in the triggers template strings... and we did use snake case when defining the sweeps macros.

We should decide what to do here, IMO since this is launch related I'd be tempted to go with the camel case style as used in triggers. Or we can change it since I don't think many people are using triggers yet. (We should chat with the folks working on model reg before doing that)

For now the one's that conflct here are:
${projectName} and ${entityName}

Reply:
In sweeps they also require that you specify envvars with ${envvar:MY_ENV_VAR} and I dropped the "envvar:" bit. I know that doesn't apply for triggers but it's one more consistency thing to call out.